test/e2e/service.go

```diff
@@ -840,6 +845,33 @@ var _ = Describe("Services", func() {
 		for _, svc := range svcs {
 			namespace := svc.Namespace
 			lbip := svc.Spec.LoadBalancerIP
+			defer func() {
```

Comment:
I think because `svc` is looping over `svcs`, you may need to pass it in as an argument to this function, as otherwise it'll just have the last element of `svcs` multiple times.